### ISSUE TITLE
Enable sccacha again

### ISSUE THREAD

**Issue by sagudev** [servo/servo#34698]
This is happening again in #34629. I think moving away from sccache-action and handling it manually using `actions/cache` and sccache's local backend is probably a better approach as it solves two issues 

1) Little benefit from sccache due to rate limiting of GHA cache service https://github.com/orgs/community/discussions/55049 & https://github.com/Mozilla-Actions/sccache-action/issues/50
2) No support for 'READ_ONLY' mode in sccache's GHA backend to avoid the 'pull_request_target' builds from corrupting base branch's cache. The [local](https://github.com/mozilla/sccache/blob/f3cf79a7f2e814c7fab80913dc1be67d1b5d3234/docs/Local.md?plain=1#L49)  and [Google Cloud Storage](https://github.com/mozilla/sccache/blob/f3cf79a7f2e814c7fab80913dc1be67d1b5d3234/docs/Gcs.md?plain=1#L7) backends have support for this mode though.

_Originally posted by @mukilan in https://github.com/servo/servo/issues/34571#issuecomment-2550312572_
            
We disabled it in https://github.com/servo/servo/pull/34692 after trying to fix it in mutiple attempts. It is suspected that new version (0.9) is causing problems. Weirdly enough mozjs repo does not have any problems.

**Comment by sagudev** [comment]
Maybe we should try older version with this:
```
- name: Run sccache-cache
  uses: mozilla-actions/sccache-action@v0.0.7
  with:
    version: "v0.7.4"
```
NOTE: after getting mozjs artifacts back this will be harder to test.

**Comment by mukilan** [comment]
> NOTE: after getting mozjs artifacts back this will be harder to test.

We could force the nightly build to always build from source, to ensure this path is always tested.

**Comment by jschwe** [comment]
Is there a noticable build time regression by removing sccache?

**Comment by sagudev** [comment]
> Is there a noticable build time regression by removing sccache?

Yes, at least in my personal fork it is noticeable. Maybe we should have just remove sccache for c++ (removing only CCACHE env var).

**Comment by jschwe** [comment]
That sounds reasonable to me

**Comment by jschwe** [comment]
Using sccache with github actions cache might also be more interesting, since it is now possible to purchase additional cache storage for 0.07$ per GiB / month (see [github actions billing](https://docs.github.com/en/billing/concepts/product-billing/github-actions)). Previously, we had lots of problems with cache trashing, since we build in many different confiugrations (multiple OS). Might be worth trying it out again with one build configuration (one os, one cargo profile), checking the cache utilization and build speed impact, and then deciding if it's still worth it.
